### DISCUSSION TITLE
SB 5.1 - OCI - Course Creation Error

### DISCUSSION THREAD

**Discussion by rjanart** [Sunbird-Ed/Community#485] [Bugs] [answered]
When we try to create a course, we are getting validation error "Metadata programs should be one of: [Nipun Bharat (FLN), Adult Education, Vocational Education, CWSN, Virtual Labs]"

{"id":"api.hierarchy.update","ver":"1.0","ts":"2023-05-10T06:57:19.689Z","params":{"resmsgid":"e820d390-eeff-11ed-999c-e79edbde7d8c","msgid":null,"status":"failed","err":"CLIENT_ERROR","errmsg":"Metadata programs should be one of: [Nipun Bharat (FLN), Adult Education, Vocational Education, CWSN, Virtual Labs]"},"responseCode":"CLIENT_ERROR","result":{}}

Here the value in the array is not program but it is verticals. 

{"code":"verticals","dataType":"list","description":"Verticals","editable":true,"inputType":"nestedselect","renderingHints":{"class":"sb-g-col-lg-1"},"label":"Verticals","name":"Verticals","placeholder":"Select Verticals","required":false,"visible":true,"range":["Nipun Bharat (FLN)","Adult Education","Vocational Education","CWSN","Virtual Labs"]},

{"code":"programs","dataType":"list","description":"Programs","editable":true,"inputType":"nestedselect","renderingHints":{"class":"sb-g-col-lg-1"},"label":"Programs","name":"Programs","placeholder":"Select Programs","required":false,"visible":true,"range":["NISHTHA 2.0","NISHTHA 3.0","Bhasha Sangam"]}

PATCH Request URL: https://diksha.gov.in/action/content/v3/hierarchy/update
{"request":{"data":{"nodesModified":{"do_3137928688602972161881":{"root":true,"objectType":"Content","metadata":{"appIcon":"","name":"TestCourse","description":"Enter description for Course","primaryCategory":"Course","additionalCategories":[],"audience":["Student"],"targetMediumIds":["z_test_framework_medium_ztestenglish"],"targetGradeLevelIds":["z_test_framework_gradelevel_ztestclass1"],"targetSubjectIds":["z_test_framework_subject_ztestmathematics"],"author":"contentcreatorprod","copyright":"ekstep","license":"CC BY 4.0","verticals":[],"programs":["NISHTHA 2.0"],"contentType":"Course","attributions":[],"copyrightYear":2022,"targetBoardIds":["z_test_framework_board_ztestcbse"],"framework":"z_test_framework","subjectIds":["z_test_framework_subject_ztestmathematics"]},"isNew":false}},"hierarchy":{"do_3137928688602972161881":{"name":"TestCourse","children":[],"root":true}},"lastUpdatedBy":"91444672-c843-4e7b-8714-5a8493aafe9a"}}}

{"id":"api.hierarchy.update","ver":"1.0","ts":"2023-05-10T06:57:19.689Z","params":{"resmsgid":"e820d390-eeff-11ed-999c-e79edbde7d8c","msgid":null,"status":"failed","err":"CLIENT_ERROR","errmsg":"Metadata programs should be one of: [Nipun Bharat (FLN), Adult Education, Vocational Education, CWSN, Virtual Labs]"},"responseCode":"CLIENT_ERROR","result":{}}

Can you please shed some light why it is validating programs against the verticals?

**Comment by anandvarada** [comment]
@amitpriyadarshi do look at this .. this look like a form configuration issue

**Comment by rjanart** [comment, marked as answer]
This issue is fixed now. In Object category Definition of the test channel "0123221758376673287017", the range of Programs have been wrong set to verticals range values.

![image](https://github.com/Sunbird-Ed/Community/assets/123344037/bc1186d6-dc9b-4598-b7d8-6e9aa9d28543)
As recommended by [maheshkumargangula](https://github.com/maheshkumargangula), category definition was corrected using Object Category Definition Update API. It works fine now(Thanks to Mahesh)

API documentation:
http://docs.sunbird.org/latest/apis/objectcategory/#operation/Update%20Object%20Category%20Definition
